libs/agno/agno/tools/dalle.py

```diff
@@ -83,18 +76,24 @@ def create_image(self, agent: Union[Agent, Team], prompt: str) -> str:
             )
             log_debug("Image generated successfully")
 
+            generated_images = []
+
             # Update the run response with the image URLs
             response_str = ""
             if response.data:
                 for img in response.data:
                     if img.url:
-                        agent.add_image(
-                            ImageArtifact(
-                                id=str(uuid4()), url=img.url, original_prompt=prompt, revised_prompt=img.revised_prompt
-                            )
+                        image_artifact = ImageArtifact(
+                            id=str(uuid4()), url=img.url, original_prompt=prompt, revised_prompt=img.revised_prompt
                         )
+                        agent.add_image(image_artifact)
```

Comment:
Ideally we won't do this anymore but automatically handle it in the model/agent. But yeh for simplicity lets keep it here for now. We can clean up in Agno 2.0